### PR TITLE
ChakraCore NuGet Package Implementation

### PR DESCRIPTION
**Pull request by pre10der89**
Adding implementation for several nuget packages that allow the ChakraCore DLLs to be referenced in .NET projects... This main tasks in git issue #85...

A nuspec has been provided for the following flavors:

ARM, x86, x64 (combined)
ARM (only)
x86 (only)
x64 (only)
Symbols [PDBs] (only)

v140 (CPP only)

### CODE REVIEW

**Comment by digitalinfinity** [comment]
@boingoing can you fix the copyright check script to handle this PR. @pre10der89, once we fix the copyright check script, you can rebase your PR on top of that change, and also update the ps1 file to change tabs to spaces.

File: Build/NuGet/.pack-version [hunk 1]
```diff
@@ -0,0 +1 @@
+2.0.2-pre
```
Comment (dilijev) [inline Build/NuGet/.pack-version]: We might want to update this version number.

**Comment by dilijev** [comment]
Hi @pre10der89, thanks so much for this!

File: Build/NuGet/package.ps1 [hunk 2]
```diff
@@ -0,0 +1,45 @@
+$root = (split-path -parent $MyInvocation.MyCommand.Definition) + '\..'
```
Comment (dilijev) [inline Build/NuGet/package.ps1]: Could you add the Microsoft copyright header to this file?

```ps1
#-------------------------------------------------------------------------------------------------------
# Copyright (C) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
#-------------------------------------------------------------------------------------------------------
```

**Comment by dilijev** [comment]
@boingoing @digitalinfinity these are the only files that weren't excluded form the check (we already have an exclusion for .nuspec files):

```
18:15:03 Check Copyright > Checking Build/NuGet/.pack-version
18:15:03 Check Copyright > Checking Build/NuGet/_._
18:15:03 Check Copyright > Checking Build/NuGet/package.ps1
```

`.pack-version` is failing because it's non-empty and doesn't have the header. We should add an exclusion for that here: https://github.com/Microsoft/ChakraCore/blob/master/jenkins/check_copyright.sh#L33

`_._` is fine because the script explicitly passes empty files.

`package.ps1` will need the copyright header added: https://github.com/Microsoft/ChakraCore/pull/2266#discussion_r93702269

File: Build/NuGet/.pack-version (hunk 1 above)
Comment (pre10der89) [inline Build/NuGet/.pack-version]: @dilijev I simply chose that value for internal testing... Is there a version you prefer?

**Comment by pre10der89** [comment]
@dilijev I don't think we can add the copyright to the pack-version file since the packaging script (as is) reads the entire content of the file and uses that for the package versioning...  

The _._ file shouldn't have the copyright either because its meant to be blank...

I was able to add the copyright to the package.ps1

The use of the .pack-version and the package.ps1 is ultimately up to your build team... I  utilize the .pack-version so that I only had to change the version in one location instead of in every NuSpec... The package.ps1 does build all the different flavors of packages, but doesn't go much beyond that...

**Comment by pre10der89** [comment]
@dilijev I have also signed the CLA twice... I see the CLA required tag is still there, however... Perhaps just a backlog...

**Comment by dilijev** [comment]
@pre10der89 We'll look into that, thanks! /cc @tcare @digitalinfinity

File: jenkins/check_copyright.sh [hunk 3]
```diff
@@ -31,6 +31,7 @@ git diff --name-only `git merge-base origin/master HEAD` HEAD |
     grep -v -E '\.filters$' |
     grep -v -E '\.targets$' |
     grep -v -E '\.nuspec$' |
+	grep -v -E '\.pack-version$' |
```
Comment (dilijev) [inline jenkins/check_copyright.sh]: Looks like you introduced a tab here ;P

File: Build/NuGet/.pack-version (hunk 1 above)
Comment (dilijev) [inline Build/NuGet/.pack-version]: @digitalinfinity what do you think? IMO it's not necessary to change this for this PR. We will update it as needed.

**Comment by pre10der89** [comment]
@dilijev Andrew Forster <anforste@bluejeansnet.com> is my primary email associated with my GitHub account

**Comment by dilijev** [comment]
@pre10der89 Apparently it was a glitch with the GitHub API Hook. It's resolved now. :)

**Comment by dilijev** [comment]
@pre10der89 Would you guys like us to release a package using these definitions for release/1.4? If so, could you retarget this pull request to release/1.4?

image

Then apply your changes onto release/1.4, and force-push that ref to the branch you used to open this PR.

**Comment by pre10der89** [comment]
@dilijev I have changed the base branch to release/1.4 and force pushed... This created a merge conflict on one of the test files... I then rebased my local branch against master and pushed again... This seems to have cleared the conflicts and it appears the CI is running... 

Let me know if I've done this right :)

**Comment by dilijev** [comment]
@pre10der89 Looks great! Thanks for your patience and indulging us on this adventure :)

**Pull request merged**